checker_test.go

```diff
@@ -0,0 +1,84 @@
+package tester
```

Comment:
I think the test file should be in a separate folder, not at the top level folder.

Reply:
Why is that?

Reply:
cleaner folder structure, test files are grouped together in a single directory

Reply:
Isn't it easier if the tests are next to the code? 

In any case, from the [Go guidelines](https://golang.org/pkg/testing/):

```
To write a new test suite, create a file whose name ends _test.go that contains the TestXxx functions as described here. Put the file in the same package as the one being tested.
```

Other examples (in and out of Segment):

* https://github.com/segmentio/centrifuge/blob/master/id_test.go
* https://github.com/segmentio/ksuid/blob/master/base62_test.go
* https://github.com/pkg/errors/blob/master/bench_test.go
* https://github.com/golang/go/blob/master/src/net/http/client_test.go